proposals/42-human-readable-mutable-names.md

```diff
@@ -0,0 +1,235 @@
+- Unfortunately, `dnslink=/ipns/` is a rare sight, due to IPNS rot, and people default to DNSLink even in cases where use of cryptographic identifiers would be a better choice.
+
+## Project definition
+#### Brief plan of attack
+
+<!--Briefly describe the milestones/steps/work needed for this project-->
+
+- Tackle IPNS and DNS fixes in parallel, in both Go and JS
+- Introduce configurable DNS resolver per TLD while adding DNS over HTTPS support to entire stack
+
+#### What does done look like?
+_What specific deliverables should completed to consider this project done?_
+
+- DNS over HTTPS is supported in js/go-ipfs/libp2p
+    - User agency around DNS is respected: every code path doing a DNS lookup (resolving DNSLink, `/dnsaddr`, `/dns*` multiaddrs) should use either explicit or implicit DNS resolvers defined in IPFS node configuration file.
+      - By default OS-level DNS resolver is used as catch-all, but the user can delegate resolution of all or specific TLDs to specific DoH resolvers.
+      - Node tries to resolve with the most specific resolver (if present for TLD), and then fall back to the global one.
+    - Visual Aid / config mock-up
+      ```json
+        "DNS": {
+          "Resolvers": {
+            "*.eth": "https://eth.link-or-cloudflare/path/to/dns-query",
+            "*.crypto": "https://unstoppablesomething.example.com/path/to/dns-query",
+            "*.libre": "https://www.opennic.org/path/to/their/dns-query", 
+            "*": "https://mozilla.cloudflare-dns.com/dns-query"
+          }
+        }
+      ```
+- DNSLink provides interop for human-readable names and enables independence from PKI and ICANN.
+    - Anyone can set up and run own naming system without our involvement, blessing or endorsement.
+    - No vendor-specific client is included. 
+- IPNS provides a reliable solution for cryptographically-verifiable mutable names that won't break, as long someone is re-publishing the latest record.
+    - IPNS re-publishing works seamlessly, out of the box, just like providing CIDs for blocks in local datastore (pinned or not).
+    - It is possible to pin IPNS address to keep it around and ensure it resolves even when original publisher is gone.
+        - Node follows IPNS record updates and keeps re-publishing latest one if original author disappears.
+        - Node follows content updates behind the IPNS record, and update content pin every time IPNS record changes. 
+            - It is possible to opt-out from this behavior to re-publish IPNS record for `en.wikipedia-on-ipfs.org` without pinning all the data.
+
+
+####  What does success look like?
+_Success means impact. How will we know we did the right thing?_
+
+<!--
+Provide success criteria. These might include particular metrics, desired changes in the types of bug reports being filed, desired changes in qualitative user feedback (measured via surveys, etc), etc.
+-->
+
+- We see % of `dnslink=/ipns/` going up.
+- We see more IPNS records on DHT
```

Comment:
Do we have these metrics today?  Should we add an item to the project plan to generate/collect this?

Reply:
I suspect we could add IPNS record metrics  to engine behind https://dht.ecosystem-dashboard.com
But this is also something for #19